core-kotlin-modules/core-kotlin-date-time/src/main/kotlin/com/baeldung/dates/datetime/DateDifference.kt

```diff
@@ -0,0 +1,48 @@
+package com.baeldung.dates.datetime
+
+import org.joda.time.format.DateTimeFormat
+import java.time.LocalDate
+import java.time.Period
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+class DateDifference {
+
+    fun dateDiffUsingJava8(fromDate: String, toDate: String): Map<Int, Int> {
+        val dateFormatter: DateTimeFormatter =  DateTimeFormatter.ofPattern("MM/dd/yyyy")
+
+        val from = LocalDate.parse(fromDate, dateFormatter)
+        val to = LocalDate.parse(toDate, dateFormatter)
+
+        val period = Period.between(from, to)
+
+        val years = period.years
+        val months = period.months
+        val days = period.days
+
+        return mapOf(
+            Calendar.YEAR to years,
+            Calendar.MONTH to months,
+            Calendar.DATE to days
+        )
```

Comment:
As a consumer of a diff between two dates I would expect a Period object instead of a map.